### PR TITLE
Add docs for Processor, Load_data

### PR DESCRIPTION
**Pull request by psteckler**
Add comments to `Processor` and `Load_data` in `Archive_lib`.

There are no code changes, so making the PR against `berkeley`.

### CODE REVIEW

**Comment by psteckler** [comment]
!ci-build-me

**Review by ymekuria — APPROVED** [review]

File: src/app/archive/lib/processor.ml [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+(* For each table in the archive database schema, there is a
+   corresponding module here containing code to read from and write to
+   that table. The module defines a type `t`, a record with fields
+   corresponding to columns in the table; typically, the `id` column
+   does not have an associated field.
+
+   The more recently-written modules use the Mina_caqti library to
+   construct the SQL for those queries. For consistency and
+   simplicity, the older modules should probably be refactored to use
+   Mina_caqti.
```
Comment (barriebyron) [inline src/app/archive/lib/processor.ml:12]:
```suggestion
(* For each table in the archive database schema, a
   corresponding module contains code to read from and write to
   that table. The module defines a type `t`, a record with fields
   corresponding to columns in the table; typically, the `id` column
   that does not have an associated field.

   The more recently written modules use the Mina_caqti library to
   construct the SQL for those queries. For consistency and
   simplicity, the older modules should probably be refactored to use
   Mina_caqti.
```

**Review by barriebyron — APPROVED** [review]
LGTM, thanks for the advice about refactoring

**Comment by psteckler** [comment]
!ci-build-me

**Review by deepthiskumar — APPROVED** [review]

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**